### PR TITLE
Remove reuseable workflows

### PR DESCRIPTION
**Pull request by nvaytet**
Remove reuseable workflows `build.yml` and `deploy.yml` to avoid having to change the `@main` version selector every time we want to make changes to the workflows.
Duplicate code instead. There is not so much code, so this is ok I think.

### CODE REVIEW

File: .github/workflows/weekly_and_release.yml [hunk 1]
```diff
@@ -7,12 +7,71 @@ on:
...
+  run:
+    shell: bash -l {0}  # required for conda env
+
 jobs:
+  formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
```
Comment (jl-wynen) [inline .github/workflows/weekly_and_release.yml]: I'm not actually sure if this is correct. I added it in my PR without thinking about release builds. The docs for auto-commit mention this only in the context or PR builds. So maybe we should remove it. (Also in `build:` below)

Comment (nvaytet) [reply .github/workflows/weekly_and_release.yml]: is it even needed for the PR builds?

Comment (jl-wynen) [reply .github/workflows/weekly_and_release.yml]: I don't really understand what it does. But the auto-commit docs say
> Note that we explicitly specify ${{ github.head_ref }} in the checkout Action. This is required in order to work with the pull_request event (or any other non-push event).

Comment (nvaytet) [reply .github/workflows/weekly_and_release.yml]: But it seems it was working fine without it... hmm

**Comment by nvaytet** [comment]
@jl-wynen did you have any more comments on this?

**Review by jl-wynen — APPROVED** [review]
No, looks good

**Pull request merged**